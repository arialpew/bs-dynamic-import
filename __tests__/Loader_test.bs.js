'use strict';

var Jest = require("@glennsl/bs-jest/src/jest.js");
var Curry = require("bs-platform/lib/js/curry.js");
var DynamicImport = require("../src/DynamicImport.bs.js");

describe("Reason", (function () {
        Jest.testPromise(/* None */0, "should work wih basic module export", (function () {
                return DynamicImport.$great$great$eq$bang(DynamicImport.$great$great$eq(DynamicImport.resolve(import("./Mocks/Reason/ReasonBase.bs.js")), (function (X) {
                                  return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](Curry._1(X[/* lazyValue */0], /* () */0) + X[/* value */1]));
                                })), (function () {
                              return Jest.fail("this case should not happen !");
                            }));
              }));
        Jest.testPromise(/* None */0, "should work with BuckleScript interop binding export", (function () {
                return DynamicImport.$great$great$eq$bang(DynamicImport.$great$great$eq(DynamicImport.resolve(import("./Mocks/Reason/ReasonJsInterop.bs.js")), (function (X) {
                                  return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](Curry._1(X[/* inc */0], 2)));
                                })), (function () {
                              return Jest.fail("this case should not happen !");
                            }));
              }));
        Jest.testPromise(/* None */0, "should work with default export", (function () {
                return DynamicImport.$great$great$eq$bang(DynamicImport.$great$great$eq(DynamicImport.resolve(import("./Mocks/Reason/ReasonDefaultExport.bs.js")), (function (X) {
                                  return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](Curry._1(X[/* lazyValue */1], /* () */0) + (X[/* value */2] + X[/* default */0])));
                                })), (function () {
                              return Jest.fail("this case should not happen !");
                            }));
              }));
        Jest.testPromise(/* None */0, "should resolve two module in parallel", (function () {
                return DynamicImport.$great$great$eq$bang(DynamicImport.$great$great$eq(DynamicImport.resolve2(/* tuple */[
                                    import("./Mocks/Reason/ReasonBase.bs.js"),
                                    import("./Mocks/Reason/ReasonJsInterop.bs.js")
                                  ]), (function (param) {
                                  var X = param[0];
                                  return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](Curry._1(param[1][/* toUpper */1], Curry._1(X[/* lazyValue */0], /* () */0) + X[/* value */1])));
                                })), (function () {
                              return Jest.fail("this case should not happen !");
                            }));
              }));
        return Jest.testPromise(/* None */0, "should fail if at least one module doesn't resolve", (function () {
                      return DynamicImport.$great$great$eq$bang(DynamicImport.$great$great$eq(DynamicImport.resolve2(/* tuple */[
                                          import("./Mocks/Reason/ReasonBase.bs.js"),
                                          import("./Mocks/Reason/???.bs.js")
                                        ]), (function () {
                                        return Jest.fail("this case should not happen !");
                                      })), (function (error) {
                                    return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](error));
                                  }));
                    }));
      }));

describe("Ocaml", (function () {
        Jest.testPromise(/* None */0, "should work wih basic module export", (function () {
                return DynamicImport.$great$great$eq$bang(DynamicImport.$great$great$eq(DynamicImport.resolve(import("./Mocks/Ocaml/OcamlBase.bs.js")), (function (X) {
                                  return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](Curry._1(X[/* lazyValue */0], /* () */0) + X[/* value */1]));
                                })), (function () {
                              return Jest.fail("this case should not happen !");
                            }));
              }));
        Jest.testPromise(/* None */0, "should work with BuckleScript interop binding export", (function () {
                return DynamicImport.$great$great$eq$bang(DynamicImport.$great$great$eq(DynamicImport.resolve(import("./Mocks/Ocaml/OcamlJsInterop.bs.js")), (function (X) {
                                  return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](Curry._1(X[/* inc */0], 2)));
                                })), (function () {
                              return Jest.fail("this case should not happen !");
                            }));
              }));
        Jest.testPromise(/* None */0, "should work with default export", (function () {
                return DynamicImport.$great$great$eq$bang(DynamicImport.$great$great$eq(DynamicImport.resolve(import("./Mocks/Ocaml/OcamlDefaultExport.bs.js")), (function (X) {
                                  return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](Curry._1(X[/* lazyValue */1], /* () */0) + (X[/* value */2] + X[/* default */0])));
                                })), (function () {
                              return Jest.fail("this case should not happen !");
                            }));
              }));
        Jest.testPromise(/* None */0, "should resolve two module in parallel", (function () {
                return DynamicImport.$great$great$eq$bang(DynamicImport.$great$great$eq(DynamicImport.resolve2(/* tuple */[
                                    import("./Mocks/Ocaml/OcamlBase.bs.js"),
                                    import("./Mocks/Ocaml/OcamlJsInterop.bs.js")
                                  ]), (function (param) {
                                  var X = param[0];
                                  return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](Curry._1(param[1][/* toUpper */1], Curry._1(X[/* lazyValue */0], /* () */0) + X[/* value */1])));
                                })), (function () {
                              return Jest.fail("this case should not happen !");
                            }));
              }));
        return Jest.testPromise(/* None */0, "should fail if at least one module doesn't resolve", (function () {
                      return DynamicImport.$great$great$eq$bang(DynamicImport.$great$great$eq(DynamicImport.resolve2(/* tuple */[
                                          import("./Mocks/Ocaml/OcamlBase.bs.js"),
                                          import("./Mocks/Ocaml/???.bs.js")
                                        ]), (function () {
                                        return Jest.fail("this case should not happen !");
                                      })), (function (error) {
                                    return Jest.Expect[/* toMatchSnapshot */16](Jest.Expect[/* expect */0](error));
                                  }));
                    }));
      }));

/*  Not a pure module */
