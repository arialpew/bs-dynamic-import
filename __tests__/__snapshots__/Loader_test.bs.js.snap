// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ocaml should fail if at least one module doesn't resolve 1`] = `[Error: Cannot find module './Mocks/Ocaml/???.bs.js' from 'Loader_test.bs.js']`;

exports[`Ocaml should resolve two module in parallel 1`] = `"AB"`;

exports[`Ocaml should work wih basic module export 1`] = `"ab"`;

exports[`Ocaml should work with BuckleScript interop binding export 1`] = `3`;

exports[`Ocaml should work with default export 1`] = `"abc"`;

exports[`Reason should fail if at least one module doesn't resolve 1`] = `[Error: Cannot find module './Mocks/Reason/???.bs.js' from 'Loader_test.bs.js']`;

exports[`Reason should resolve two module in parallel 1`] = `"AB"`;

exports[`Reason should work wih basic module export 1`] = `"ab"`;

exports[`Reason should work with BuckleScript interop binding export 1`] = `3`;

exports[`Reason should work with default export 1`] = `"abc"`;
